src/errors.ts

```diff
@@ -51,7 +51,7 @@ const isZodFastifySchemaValidationError = (
 
 export const hasZodFastifySchemaValidationErrors = (
   error: unknown,
-): error is FastifyError & { validation: ZodFastifySchemaValidationError[] } =>
+): error is Omit<FastifyError, 'validation'> & { validation: ZodFastifySchemaValidationError[] } =>
```

Comment:
can you add test that would have been failing before?

Reply:
Hmm weird, in the project itself it is not failing. (I used the code before the change + I am on the same typescript version)

![image](https://github.com/user-attachments/assets/6e52b3f0-8768-4e95-827e-695f00d930bc)

Reply:
![image](https://github.com/user-attachments/assets/5cdda1c8-4a0c-4e65-81ee-825d2bc2f732)

Reply:
This is weird in typescript:

Look at this: (in the same personal project)
![image](https://github.com/user-attachments/assets/5986ddc7-f12f-40bb-8c12-93de10288d2a)

Versus:
![image](https://github.com/user-attachments/assets/bd32d739-b47a-44eb-be48-bd8337ee8072)

Reply:
I will write some test for both then.

Reply:
`forEach` gives a different type than `[0]`